src/inmanta/data/__init__.py

```diff
@@ -775,21 +793,80 @@ def get_default_value(self) -> T:
 
     default_value = property(get_default_value)
 
-    def is_unique(self) -> bool:
-        return self._unique
-
-    unique = property(is_unique)
-
-    def is_reference(self) -> bool:
-        return self._reference
-
-    reference = property(is_reference)
+    @property
+    def ignore(self) -> bool:
+        return self._ignore
 
     def is_part_of_primary_key(self) -> bool:
         return self._part_of_primary_key
 
     part_of_primary_key = property(is_part_of_primary_key)
 
+    @property
+    def is_many(self) -> bool:
+        return self._is_many
+
+    def _validate_single(self, name: str, value: object) -> object:
+        """Validate a single value against the types in this field."""
+        if not (value.__class__ is self.field_type or isinstance(value, self.field_type)):
+            raise TypeError(
+                "Field %s should have the correct type (%s instead of %s)"
+                % (name, self.field_type.__name__, type(value).__name__)
+            )
+        return value
+
+    def validate(self, name: str, value: object) -> object:
```

Comment:
I think we could provide a bit more flexible guarantees for these methods but I'm not sure if it would actually gain us anything. They always seem to return the same object so we could type it as `..., value: T) -> T`. That said, I'm wondering why these even return anything at all, there's no conversion going on like with pydantic.

I understand this is copied from inmanta-lsm, it's fine as is for me, just thought you might appreciate the speculation.

Reply:
I removed the return type and updated the use. It indeed makes no sense.